main.go

```diff
@@ -1,94 +0,0 @@
-// time-stamp microservice receives timestamps and returns json form of natural/unix timestamp.
-// Exercise by Free Code Camp.
-package main
-
-import (
-	"encoding/json"
-	"fmt"
-	"net/http"
-	"os"
-	"strconv"
-	"strings"
-	"time"
-)
-
-// Date struct returned as json.
-type Date struct {
-	Unix    *int64  `json:"unix"`
-	Natural *string `json:"natural"`
-}
-
-func main() {
-	http.HandleFunc("/", timestamp)
-
-	server := http.Server{
-		Addr: ":" + os.Getenv("PORT"),
-	}
-	fmt.Println("Listening on [", server.Addr, "]...")
-	err := server.ListenAndServe()
-	if err != nil {
-		panic(err)
-	}
-}
-
-func timestamp(w http.ResponseWriter, r *http.Request) {
-
-	tstr := r.URL.Path[1:]
-
-	w.Header().Set("Content-Type", "application/json")
-	w.Header().Set("charset", "utf-8")
-
-	t, err := returnTime(tstr)
-	if err != nil {
-		nilJSON, err := json.Marshal(Date{nil, nil})
-		if err != nil {
-			http.Error(w, "Json string failed to write", 500)
-			return
-		}
-		w.Write(nilJSON)
-		return
-	}
-
-	naturalDate := fmt.Sprintf("%v %v, %v", t.Month().String(), t.Day(), t.Year())
-	naturalTime := t.Unix()
-
-	date := Date{
-		Unix:    &naturalTime,
-		Natural: &naturalDate}
-
-	responseJSON, err := json.Marshal(date)
-	if err != nil {
-		http.Error(w, "Json string failed to write", 500)
-		return
-	}
-
-	w.Write(responseJSON)
```

Comment:
You're being a bit verbose here. because `w` is an io.Writer, you can use json.Encoder instead of json.Marshal followed by w.Write.

`if err := json.NewEncoder(w).Encode(&date); err != nil {`